### PR TITLE
Show why configuration cache test is ignored

### PR DESCRIPTION
**Pull request by lptr**
Mention the reason in the exception thrown.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test and merge.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithConfigurationCacheRule.groovy [hunk 1]
```diff
@@ -37,23 +36,22 @@ class UnsupportedWithConfigurationCacheRule implements TestRule {
...
-        SkippingRuleStatement(Statement next) {
-            this.next = next
+        SkippingRuleStatement(String reason) {
+            this.reason = reason
         }
 
         @Override
         void evaluate() throws Throwable {
-            assumeFalse(GradleContextualExecuter.isConfigCache())
-            next.evaluate()
```
Comment (bamboo) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithConfigurationCacheRule.groovy:56]: 🤔 The intention is that the test still runs when `!GradleContextualExecuter.configCache`, right? Does this change preserve that behavior?

Comment (bamboo) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithConfigurationCacheRule.groovy:56]: Yeah, I see that `apply` above actually already checks for `isNotConfigCache`. Nothing to see here...

**Review by bamboo — APPROVED** [review]

**Comment by bamboo** [comment]
@bot-gradle test and merge

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithConfigurationCacheRule.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithConfigurationCacheRule.groovy:56]: Yep, `assumeFalse()` would always throw, so we'd never get to `next.evaluate()`.

**Pull request merged**